File: backend/expense_tracker_backend/src/main/kotlin/mk/sorsix/com/expense_tracker_backend/domain/DailyPlan.kt
package mk.sorsix.com.expense_tracker_backend.domain

import jakarta.persistence.*
import java.math.BigDecimal
import java.time.LocalDate

@Entity
@Table(name = "daily_plan")
data class DailyPlan(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long = 0,

    @ManyToOne
    @JoinColumn(name = "plan_id")
    val plan: Plan = Plan(),

    val date: LocalDate = LocalDate.now(),
    val allocatedAmount: BigDecimal = BigDecimal.ZERO
)

File: frontend/src/app/models/daily-plan.ts
export interface DailyPlan {
  id: number;
  planId: number;
  date: string;
  allocatedAmount: number;
  actualSpent: number;
}

export interface DailyPlanRequest {
  date: string;
  allocatedAmount: number;
  confirmOverBudget?: boolean;
}
